sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStoreErrors.scala

```diff
@@ -435,15 +452,31 @@ class StateStoreFailedToGetChangelogWriter(version: Long, e: Throwable)
     messageParameters = Map("version" -> version.toString),
     cause = e)
 
-class StateStoreKeyRowFormatValidationFailure(errorMsg: String)
+class StateStoreKeyRowFormatValidationFailure(errorMsg: String, stateStoreID: String)
   extends SparkRuntimeException(
     errorClass = "STATE_STORE_KEY_ROW_FORMAT_VALIDATION_FAILURE",
-    messageParameters = Map("errorMsg" -> errorMsg))
+    messageParameters = Map("errorMsg" -> errorMsg, "stateStoreID" -> stateStoreID))
+  with ConvertableToCannotLoadStoreError {
+    override def convertToCannotLoadStoreError(): SparkException = {
+      new SparkException(
+        errorClass = "CANNOT_LOAD_STATE_STORE.KEY_ROW_FORMAT_VALIDATION_FAILURE",
+        messageParameters = Map("errorMsg" -> errorMsg, "stateStoreID" -> stateStoreID),
```

Comment:
instead of duplicating params and error message here. You can just call `this.message` which contains the full message from the previous error class, and then you can pass it into the new exception object here